pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java

```diff
@@ -54,4 +54,9 @@ public List<RexExpression> getAggCalls() {
   public List<RexExpression> getGroupSet() {
     return _groupSet;
   }
+
+  @Override
+  public String debugString() {
```

Comment:
In the existing engine, we implement `toExplainString` at the physical operator level and each operator also has a EXPLAIN_NAME. 

Even in this case, I think the API could be called `explainString` or something similar even though we are doing at the logical node level and not at the operator level.

Reply:
Ohh I just noticed this is not wired into the EXPLAIN PLAN. We do support EXPLAIN for multi-stage btw. We can probably also do this by wiring this up with EXPLAIN PLAN for multi-stage (can do as a follow up as well).

 The output in the description looks reasonably good

Reply:
can we also add instances / worker assignment into the output ?

Reply:
@siddharthteotia you read my mind 😉 that's _just_ what I was working on (I'll update PR description with examples once I've worked out the bugs).